pyccel/codegen/printing/ccode.py

```diff
@@ -2184,6 +2187,14 @@ def _print_Assign(self, expr):
         rhs = self._print(expr.rhs)
         return prefix_code+'{} = {};\n'.format(lhs, rhs)
 
+    def _print_SetPop(self, expr):
+        var_type = self.get_declare_type(expr.set_variable)
+        element_type = self.get_c_type(expr.set_variable.class_type.element_type)
+        self.add_import(Import(f'Set_pop_macro/{var_type}/{element_type}', Module(f'Set_pop_macro/{var_type}/{element_type}', (), ())))
+        self.add_import(Import('STC_Extensions/Set_extensions', Module('STC_Extensions/Set_extensions', (), ())))
+        set_var = expr.set_variable.name
+        return f'{var_type}_pop(&{set_var})'
```

Comment:
Please use `self._print` to print objects as [documented](https://github.com/pyccel/pyccel/blob/devel/developer_docs/codegen_stage.md#_print)

This code will break for sets in classes. It will also be unnecessarily verbose for a set which is an optional argument (e.g. `set_int_pop(&(*set_var))`). You should use `ObjectAddress` to help with this

Reply:
Fixed : db9e6c0